File: 백준/Silver/1072. 게임/게임.py
# import math # 소수점 버림 함수

x, y = map(int, input().split())
# z = math.trunc(y / x * 100)
z = y * 100 // x

# 예외 설정, 한 번이라도 졌으면 100 안 됨
# if z >= 99:
#     print(-1)

start = 1
end = 1000000000
result = -1

while start <= end:
    mid = (start + end) // 2
    # if math.trunc((y+mid) / (x+mid) * 100) > z: # 승률이 올랐다면
    if ((y+mid) * 100 // (x+mid)) > z:
        end = mid-1
        result = mid
    else:
        start = mid+1
    
print(result)

File: 백준/Silver/1072. 게임/게임.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int x = Integer.parseInt(st.nextToken());
        int y = Integer.parseInt(st.nextToken());
        int z = (int) ((long) y * 100 / x);
        
        int start = 1; // 추가로 이긴 게임 수
        int end = 1_000_000_000;
        int result = -1;
        
        while (start <= end) {
            int mid = (start + end) / 2;
            
            int nx = x + mid;
            int ny = y + mid;
            double nz = (int) ((long) ny * 100 / nx);
            
            if (z < nz) {
                result = mid;
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        
        System.out.println(result);
    }
}
